File: CS/MsaglHelpers/LayoutCalculators/PhyloTreeLayoutCalculator.cs
using Microsoft.Msagl.Core.Layout;
using Microsoft.Msagl.Layout.Layered;
using Microsoft.Msagl.Core.Geometry.Curves;
using Microsoft.Msagl.Core.Routing;
using System;
using Microsoft.Msagl.Prototype.Phylo;

namespace MsaglHelpers {
    public class PhyloTreeLayoutCalculator : ILayoutCalculator {
        public LayoutAlgorithmSettings LayoutAlgorithmSettings {
            get {
                return new SugiyamaLayoutSettings() {
                    Transformation = PlaneTransformation.Rotation(Math.PI),
                    EdgeRoutingSettings = {
                        EdgeRoutingMode =  EdgeRoutingMode.StraightLine,
                    }
                };
            }
        }
        public void CalculateLayout(GeometryGraph phyloTree) {
            Microsoft.Msagl.Miscellaneous.LayoutHelpers.CalculateLayout(phyloTree, LayoutAlgorithmSettings, null);
        }
    }
}

File: VB/MsaglHelpers/LayoutCalculators/PhyloTreeLayoutCalculator.vb
Imports Microsoft.Msagl.Core.Layout
Imports Microsoft.Msagl.Layout.Layered
Imports Microsoft.Msagl.Core.Geometry.Curves
Imports Microsoft.Msagl.Core.Routing
Imports System
Imports Microsoft.Msagl.Prototype.Phylo

Namespace MsaglHelpers
    Public Class PhyloTreeLayoutCalculator
        Implements ILayoutCalculator

        Public ReadOnly Property LayoutAlgorithmSettings() As LayoutAlgorithmSettings Implements ILayoutCalculator.LayoutAlgorithmSettings
            Get
                Return New SugiyamaLayoutSettings() With {
                    .Transformation = PlaneTransformation.Rotation(Math.PI), .EdgeRoutingSettings = New EdgeRoutingSettings() With {.EdgeRoutingMode = EdgeRoutingMode.StraightLine}
                }
            End Get
        End Property
        Public Sub CalculateLayout(ByVal phyloTree As GeometryGraph) Implements ILayoutCalculator.CalculateLayout
            Microsoft.Msagl.Miscellaneous.LayoutHelpers.CalculateLayout(phyloTree, LayoutAlgorithmSettings, Nothing)
        End Sub
    End Class
End Namespace
